fix(backup): backup import fails if the community was left

jrainville opened the pull request:
This is a fix for the backup and also just a performance improvement. Basically, when we have a SyncCommunity message, we leave the community if the state of said community is not pending or joined (meaning we left it). However, in the case of backup messages, we backup left communities too, but when trying to "leave" those communities, it threw an error about unknown chat, since we never spectated that community in that imported profile.

The solution is just to early return when we don't have anything to leave. We gracefully ignore the error in the Sync function and pass the error when actively trying to leave (you can't leave twice).

Added tests and also cleaned up some warnings.

This will also happen in release, so I think I'll need to cherry-pick it?

jrainville committed:
fix(backup): backup import fails if the community was left

This is a fix for the backup and also just a performance improvement. Basically, when we have a SyncCommunity message, we leave the community if the state of said community is not pending or joined (meaning we left it). However, in the case of backup messages, we backup left communities too, but when trying to "leave" those communities, it threw an error about unknown chat, since we never spectated that community in that imported profile.

The solution is just to early return when we don't have anything to leave. We gracefully ignore the error in the Sync function and pass the error when actively trying to leave (you can't leave twice).

Added tests and also cleaned up some warnings.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (38)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 4ae910df | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6880/1/) | 2025-08-28 20:38:07 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6880/1/consoleText) |
| :heavy_check_mark: | 4ae910df | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6880/1/) | 2025-08-28 20:39:24 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250828-203616-4ae910-pr6880.zip) |
| :x: | 4ae910df | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6880/1/) | 2025-08-28 20:40:28 | ~4 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6880/1/consoleText) |
| :heavy_check_mark: | 4ae910df | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6880/1/) | 2025-08-28 20:40:40 | ~4 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250828-203622-4ae910-pr6880.zip) |
| :heavy_check_mark: | 4ae910df | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6880/1/) | 2025-08-28 20:42:47 | ~6 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250828-203628-4ae910-pr6880.zip) |
| :heavy_check_mark: | 4ae910df | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/1/) | 2025-08-28 20:45:19 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/1//consoleText) |
| :heavy_check_mark: | 4ae910df | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6880/1/) | 2025-08-28 20:46:52 | ~10 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250828-203618-4ae910-pr6880.zip) |
| :heavy_check_mark: | 4ae910df | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/1/) | 2025-08-28 21:01:10 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/1//consoleText) |
| | | | | | | |
| :x: | 8ffe9e6c | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6880/2/) | 2025-08-28 20:39:15 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6880/2/consoleText) |
| :heavy_check_mark: | 8ffe9e6c | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6880/2/) | 2025-08-28 20:42:27 | ~2 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250828-203927-8ffe9e-pr6880.zip) |
| :x: | 8ffe9e6c | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6880/2/) | 2025-08-28 20:44:01 | ~3 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6880/2/consoleText) |
| :heavy_check_mark: | 8ffe9e6c | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6880/2/) | 2025-08-28 20:44:11 | ~3 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250828-204045-8ffe9e-pr6880.zip) |
| :heavy_check_mark: | 8ffe9e6c | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6880/2/) | 2025-08-28 20:49:16 | ~6 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250828-204250-8ffe9e-pr6880.zip) |
| :heavy_check_mark: | 8ffe9e6c | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/2/) | 2025-08-28 20:53:54 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/2//consoleText) |
| :heavy_check_mark: | 8ffe9e6c | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6880/2/) | 2025-08-28 20:57:08 | ~10 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250828-204656-8ffe9e-pr6880.zip) |
| :heavy_check_mark: | 8ffe9e6c | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/2/) | 2025-08-28 21:25:33 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/2//consoleText) |
| | | | | | | |
| :x: | 1f4fee31 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6880/3/) | 2025-08-29 15:20:13 | ~4 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6880/3/consoleText) |
| :x: | 1f4fee31 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6880/3/) | 2025-08-29 15:21:09 | ~5 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6880/3/consoleText) |
| :heavy_check_mark: | 1f4fee31 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6880/3/) | 2025-08-29 15:21:57 | ~5 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250829-151606-1f4fee-pr6880.zip) |
| :heavy_check_mark: | 1f4fee31 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6880/3/) | 2025-08-29 15:22:38 | ~6 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250829-151612-1f4fee-pr6880.zip) |
| :heavy_check_mark: | 1f4fee31 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6880/3/) | 2025-08-29 15:23:53 | ~7 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250829-151559-1f4fee-pr6880.zip) |
| :heavy_check_mark: | 1f4fee31 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/3/) | 2025-08-29 15:31:41 | ~15 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/3//consoleText) |
| :heavy_check_mark: | 1f4fee31 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6880/3/) | 2025-08-29 15:33:36 | ~17 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250829-151601-1f4fee-pr6880.zip) |
| :heavy_check_mark: | 1f4fee31 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/3/) | 2025-08-29 15:47:37 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/3//consoleText) |
| | | | | | | |
| :x: | 63e4d39a | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6880/4/) | 2025-08-29 15:47:34 | ~4 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6880/4/consoleText) |
| :heavy_check_mark: | 63e4d39a | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6880/4/) | 2025-08-29 15:47:42 | ~4 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250829-154328-63e4d3-pr6880.zip) |
| :x: | 63e4d39a | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6880/4/) | 2025-08-29 15:49:32 | ~6 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6880/4/consoleText) |
| :heavy_check_mark: | 63e4d39a | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6880/4/) | 2025-08-29 15:49:40 | ~6 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250829-154334-63e4d3-pr6880.zip) |
| :heavy_check_mark: | 63e4d39a | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6880/4/) | 2025-08-29 15:52:23 | ~9 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250829-154323-63e4d3-pr6880.zip) |
| :heavy_check_mark: | 63e4d39a | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6880/4/) | 2025-08-29 15:59:38 | ~16 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250829-154324-63e4d3-pr6880.zip) |
| :heavy_check_mark: | 63e4d39a | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/4/) | 2025-08-29 15:59:51 | ~16 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/4//consoleText) |
| :heavy_check_mark: | 63e4d39a | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/4/) | 2025-08-29 16:16:06 | ~28 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | dfca3960 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6880/5/) | 2025-09-02 15:23:36 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250902-152006-dfca39-pr6880.zip) |
| :heavy_check_mark: | dfca3960 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6880/5/) | 2025-09-02 15:25:31 | ~5 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250902-152010-dfca39-pr6880.zip) |
| :heavy_check_mark: | dfca3960 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6880/5/) | 2025-09-02 15:26:26 | ~6 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250902-152016-dfca39-pr6880.zip) |
| :heavy_check_mark: | dfca3960 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/5/) | 2025-09-02 15:28:46 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/5//consoleText) |
| :heavy_check_mark: | dfca3960 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6880/5/) | 2025-09-02 15:29:42 | ~9 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250902-152008-dfca39-pr6880.zip) |
| :heavy_check_mark: | dfca3960 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/5/) | 2025-09-02 15:44:16 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/5//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 065ef36a | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6880/6/) | 2025-09-02 15:53:38 | ~2 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250902-155039-065ef3-pr6880.zip) |
| :heavy_check_mark: | 065ef36a | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6880/6/) | 2025-09-02 15:55:34 | ~4 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250902-155044-065ef3-pr6880.zip) |
| :heavy_check_mark: | 065ef36a | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6880/6/) | 2025-09-02 15:57:01 | ~6 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250902-155049-065ef3-pr6880.zip) |
| :heavy_multiplication_x: | 065ef36a | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/6/) | 2025-09-02 15:58:51 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/6//consoleText) |
| :heavy_check_mark: | 065ef36a | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6880/6/) | 2025-09-02 16:00:37 | ~9 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250902-155040-065ef3-pr6880.zip) |
| :heavy_check_mark: | 065ef36a | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/6/) | 2025-09-02 16:15:46 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/6//consoleText) |
| :heavy_multiplication_x: | 065ef36a | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/7/) | 2025-09-02 18:23:22 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 5d38d089 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6880/7/) | 2025-09-02 18:57:14 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250902-185413-5d38d0-pr6880.zip) |
| :heavy_check_mark: | 5d38d089 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6880/7/) | 2025-09-02 18:57:19 | ~2 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250902-185419-5d38d0-pr6880.zip) |
| :heavy_check_mark: | 5d38d089 | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6880/7/) | 2025-09-02 19:00:46 | ~6 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250902-185424-5d38d0-pr6880.zip) |
| :heavy_multiplication_x: | 5d38d089 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/8/) | 2025-09-02 19:02:11 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/8//consoleText) |
| :heavy_check_mark: | 5d38d089 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6880/7/) | 2025-09-02 19:04:56 | ~10 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250902-185414-5d38d0-pr6880.zip) |
| :heavy_check_mark: | 5d38d089 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/9/) | 2025-09-02 19:17:31 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6880/9//consoleText) |
| :heavy_check_mark: | 5d38d089 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/7/) | 2025-09-02 19:19:31 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6880/7//consoleText) |

jrainville commented:
@igor-sirotin I requested review again because I updated the PR/commit description with an accurate description. The problem was no spectated communities, but left ones.

I also added tests and cleaned some warnings

jrainville committed:
chore: skip flaky test_restored_on_existing_restored_account_fails test
